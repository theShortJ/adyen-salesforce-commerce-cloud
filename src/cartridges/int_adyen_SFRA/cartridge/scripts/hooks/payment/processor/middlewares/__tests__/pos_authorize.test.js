/* eslint-disable global-require */
let pos_authorize;
let orderNumber;
let paymentInstrument;
let paymentProcessor;
let Logger;

beforeEach(() => {
  pos_authorize = require('../pos_authorize');
  orderNumber = 'mockedNum';
  paymentInstrument = {
    paymentTransaction: {},
  };
  paymentProcessor = 'mockedPaymentProcessor';
  Logger = require('dw/system/Logger');
  jest.clearAllMocks();
});

afterEach(() => {
  jest.resetModules();
});

describe('pos_authorize', () => {
  it('should return error if there is no terminal ID', () => {
    const { getForm } = require('server').forms;
    getForm.mockImplementation(() => ({
      adyenPaymentFields: {
        terminalId: { value: null },
      },
    }));

    const authorizeResult = pos_authorize(
      orderNumber,
      paymentInstrument,
      paymentProcessor,
    );
    expect(authorizeResult).toMatchSnapshot();
    expect(Logger.error.mock.calls).toMatchSnapshot();
  });

  it('should return error if createTerminalPayment fails', () => {
    const {
      createTerminalPayment,
    } = require('*/cartridge/scripts/adyenTerminalApi');
    createTerminalPayment.mockImplementation(() => ({
      error: true,
      response: 'mockedResponse',
    }));

    const authorizeResult = pos_authorize(
      orderNumber,
      paymentInstrument,
      paymentProcessor,
    );
    expect(authorizeResult).toMatchSnapshot();
    expect(Logger.error.mock.calls).toMatchSnapshot();
  });

  it('should return success response when createTerminalPayment passes', () => {
    const authorizeResult = pos_authorize(
      orderNumber,
      paymentInstrument,
      paymentProcessor,
    );
    expect(authorizeResult).toMatchSnapshot();
  });
});
