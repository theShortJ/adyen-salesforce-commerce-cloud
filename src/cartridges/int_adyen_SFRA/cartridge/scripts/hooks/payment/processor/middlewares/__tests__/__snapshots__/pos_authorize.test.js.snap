// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pos_authorize should return error if createTerminalPayment fails 1`] = `
Object {
  "authorized": false,
  "error": true,
  "fieldErrors": Array [],
  "serverErrors": Array [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;

exports[`pos_authorize should return error if createTerminalPayment fails 2`] = `
Array [
  Array [
    "POS Authorise error, result: mockedResponse",
  ],
]
`;

exports[`pos_authorize should return error if there is no terminal ID 1`] = `
Object {
  "authorized": false,
  "error": true,
  "fieldErrors": Array [],
  "serverErrors": Array [
    "mocked_error.payment.processor.not.supported",
  ],
}
`;

exports[`pos_authorize should return error if there is no terminal ID 2`] = `
Array [
  Array [
    "No terminal selected",
  ],
]
`;

exports[`pos_authorize should return success response when createTerminalPayment passes 1`] = `
Object {
  "response": "mockedSuccessResponse",
}
`;
